cmd/gossamer: export toml congfiguration file

ryanchristo opened the pull request:
## Changes
- export toml congfiguration file using the `export` subcommand
- add `--force` flag option to skip confirmation prompts
- use lowercase unless proper name for cmd and dot logs
- fix `newTextContext` to include setting flag values

## Tests:

```
make gossamer

./bin/gossamer --config test.toml export
./bin/gossamer --config test.toml --name TESTNODE export

./bin/gossamer --config test.toml export --force

./bin/gossamer init
./bin/gossamer init --force
```

### Issues:
closes #686, related to #669

ryanchristo commented on cmd/gossamer/export.go:
```diff
@@ -33,43 +34,31 @@ func exportAction(ctx *cli.Context) error {
-			}
-		}()
-	}
+		// TODO: confirm once #767 is merged
```
waiting for #767 to be merged, then will include the following code:

```
log.Warn(
	"[cmd] toml configuration file already exists",
	"config", config,
)

if confirmMessage("Would you like to overwrite the existing file? [Y/n]") {

	cfg := createExportConfig(ctx)
	
	file := dot.ExportConfig(cfg, config)
	// export config will exit and log error on error

	log.Info("[cmd] Exported toml configuration file", "path", file.Name())
}

return nil

ryanchristo commented:
cc @noot this one needs a review as well :pray:

ryanchristo commented:
but it can also wait for #767

ryanchristo commented on cmd/gossamer/config.go:
```diff
@@ -31,6 +31,9 @@ import (
 )
 
+// DefaultCfg is the default configuration
+var DefaultCfg = dot.GssmrConfig() // TODO: investigate default node other than gssmr #776
```
#776

ryanchristo commented on dot/utils.go:
```diff
@@ -108,7 +109,7 @@ func NewTestGenesisFile(t *testing.T, cfg *Config) *os.File {
 	require.Nil(t, err)
 
-	fp := getGssmrGenesisPath(t)
+	fp := getGssmrGenesisPath(t) // TODO: investigate default node other than gssmr #776
```
#776

ryanchristo commented on dot/utils.go:
```diff
@@ -163,6 +164,8 @@ func NewTestGenesisAndRuntime(t *testing.T) string {
 	return genFile.Name()
 }
 
+// TODO: investigate default node other than gssmr #776
```
#776

ryanchristo commented:
cc @noot @edwardmack @thomasmodeneis ready for review when you get a chance

noot commented:
@ryanchristo  looks good, is there a way to get `./bin/gossamer init --force` to work along with `./bin/gossamer --force init`? maybe with the FixFlagOrder func?

noot commented on cmd/gossamer/flags.go:
```diff
@@ -171,32 +176,21 @@ var (
 	// GlobalFlags are flags that are valid for use with all commands
 	GlobalFlags = []cli.Flag{
 		VerbosityFlag,
+		ForceFlag,
```
hmm, not sure if this is global, since it's not valid with just `./bin/gossamer`

ryanchristo replied:
Not at the moment but I figured it was a global flag in the sense that we use it whenever there is a confirm prompt even though we are not currently using it in the root command.

ryanchristo replied:
I can move it but it seems global to me. Where would you recommend I move it to?

noot replied:
I think we should only make it global when necessary, if you could move it to InitFlags and ExportFlags that'd be good

ryanchristo commented:
cc @noot yea I was noticing the `FixFlagOrder` could be optimized, would be nice to open a separate issue but if you have a quick fix in mind, I would be happy to include here.

noot commented:
does the FixFlagOrder not work for this PR on --force? if not that's ok, a separate issue is fine

ryanchristo commented:
ok @noot, `--force` is no longer global therefore `init --force` is now working but since it is no longer global `--force init` will not work. Let me know if this addresses your comments.
